MIGRATION_GUIDE.md

```diff
@@ -1,14 +1,264 @@
-# [WIP] N-API Migration Guide
+# N-API Migration Guide
 
-## Context
+## Getting started
+
+### Supported Node versions
+
+The N-API backend of Neon requires a minimum Node version of 10.0.
+
+### Enabling the N-API backend
+
+To enable the N-API backend, you need to enable the `"napi"` feature flag in your `Cargo.toml`:
+
+```toml
+[dependencies.neon]
+version = "0.7.1"
+default-features = false
+features = ["napi"]
+```
+
+## Minor API changes
+
+### Context
 
 Many methods that previously did not require context (e.g., `JsString::size`) now require a context. In many cases, this means adding an additional argument or using a convenience method on the `Context` trait.
 
-### Impacted methods
+#### Affected methods
+
+##### Handle
 
+* `Handle`
+    - `is_a`
+    - `downcast`
+
+`Handle::downcast` also requires a second type argument for the context type. This can usually be inferred, so you can typically use `_`.
+
+**Before:**
+```rust
+value.downcast::<JsNumber>()
+```
+
+**After:**
+
+```rust
+value.downcast::<JsNumber, _>(&mut cx)
+```
+
+##### Primitive types
+
+* `JsBoolean`
+    - `value`
+* `JsNull`
+    - `new`
 * `JsString`
     - `size`
     - `value`
+* `JsNumber`
+    - `value`
+* `JsUndefined`
+    - `new`
+
+##### Object APIs
+
 * `PropertyKey`
     - `get_from`
     - `set_from`
+
+### Handle equality
+
+Handles no longer implement `Eq` or `PartialEq`, which had underspecified behavior. Use `Value::strict_equals` instead to invoke the behavior of JavaScript's `===` operator.
+
+## Major API changes
+
+The N-API backend introduces two categories of significant change:
+
+1. Embedding Rust data, which is no longer done through the awkward and complex `declare_types!` (i.e. classes) macro, but through a simpler primitive: the `JsBox` API.
+2. Concurrency, which is offered through the Event Queue API instead of the Task API or Event Handlers, both of which are deprecated and removed in the N-API backend.
+
+### Embedding Rust data
+
+The `declare_types!` macro is deprecated and replaced by the `JsBox` type.
+
+_Rationale:_ The `declare_types!` macro provides a syntax for defining classes, but requires substantial boilerplate and is unergonomic for simple cases. It's also not flexible enough to express the full range of JavaScript classes syntax and semantics. With the `JsBox` type, it's easy to embed Rust data in JavaScript objects, which can then be nested inside of more feature-rich classes defined in pure JavaScript (or TypeScript).
```

Comment:
It might also be worth mentioning that IDEs struggle with code inside macros. The `declare_types` macro placed large amounts of code inside a macro reducing IDE ergonomics.